litecommands-core/src/dev/rollczi/litecommands/argument/resolver/standard/DurationArgumentResolver.java

```diff
@@ -0,0 +1,35 @@
+package dev.rollczi.litecommands.argument.resolver.standard;
+
+import dev.rollczi.litecommands.argument.Argument;
+import dev.rollczi.litecommands.argument.resolver.ArgumentResolver;
+import dev.rollczi.litecommands.argument.parser.ParseResult;
+import dev.rollczi.litecommands.shared.FailedReason;
+import dev.rollczi.litecommands.invalidusage.InvalidUsage;
+import dev.rollczi.litecommands.invocation.Invocation;
+import dev.rollczi.litecommands.suggestion.SuggestionContext;
+import dev.rollczi.litecommands.suggestion.SuggestionResult;
+import dev.rollczi.litecommands.time.DurationParser;
+
+import java.time.Duration;
+
+public class DurationArgumentResolver<SENDER> extends ArgumentResolver<SENDER, Duration> {
+
+    @Override
+    protected ParseResult<Duration> parse(Invocation<SENDER> invocation, Argument<Duration> context, String argument) {
+        try {
+            Duration parse = DurationParser.DATE_TIME_UNITS.parse(argument);
+
+            return ParseResult.success(parse);
+        }
+        catch (IllegalArgumentException e) {
+            return ParseResult.failure(InvalidUsage.Cause.INVALID_ARGUMENT);
+        }
+    }
+
+    @Override
+    public SuggestionResult suggest(Invocation<SENDER> invocation, Argument<Duration> argument, SuggestionContext context) {
+        return SuggestionResult.of("1d"); // You can customize suggestions if needed
```

Comment:
remove `// You can customize suggestions if needed`
add more suggestions similarly to [this](https://github.com/Rollczi/LiteCommands/blob/v2.x.x/litecommands-core/src/main/java/dev/rollczi/litecommands/argument/basictype/time/DurationArgument.java#L11) (still as string)